### PR TITLE
Configurable image pull secrets

### LINKED ISSUE #517: Make docker credentials optional
When deployed to an environment not in NAV, the docker pull secrets does not exist. This results in errors being logged during deployment.

```
[2024-04-10T13:46:43.657891447Z] Status: in_progress: Pod/datadoc-pr-281-749f749699-4mnrg (FailedToRetrieveImagePullSecret): Unable to retrieve some image pull secrets (gar-docker-credentials); attempting to pull the image may not succeed
```

The secret is currently defined here:

https://github.com/nais/naiserator/blob/0e302759711f9fb840ae0f74aa591946ac0f3a9a/pkg/resourcecreator/pod/pod.go#L105-L108

Let's either make this optional, or configurable through Fasit

### PR DESCRIPTION
**Pull request by thokra-nav**
Make image pull secrets configurable. Fixes #517

### CODE REVIEW

File: pkg/resourcecreator/testdata/naisjob/vanilla_job_on_premises.yaml [hunk 1]
```diff
@@ -123,7 +123,7 @@ tests:
                       capabilities:
                         drop: ["ALL"]
                       seccompProfile:
-                          type: RuntimeDefault
```
Comment (kimtore) [inline pkg/resourcecreator/testdata/naisjob/vanilla_job_on_premises.yaml:126]: Interesting... how did this ever work?

File: pkg/resourcecreator/testdata/vanilla_legacy_gcp_w_pulll_secrets.yaml
Comment (kimtore) [inline pkg/resourcecreator/testdata/vanilla_legacy_gcp_w_pulll_secrets.yaml:1]: How about just testing for the `Deployment` resource, as the other resources are handled by other tests?

File: pkg/resourcecreator/testdata/vanilla_on_premises_w_pull_secrets.yaml
Comment (kimtore) [inline pkg/resourcecreator/testdata/vanilla_on_premises_w_pull_secrets.yaml:1]: Same as above.

**Review by kimtore — APPROVED** [review]
LGTM except for test files, see comments. Nice work.

**Review by kimtore — CHANGES_REQUESTED** [review]
wrong button :P

File: pkg/resourcecreator/testdata/naisjob/vanilla_job_on_premises.yaml (hunk 1 above)
Comment (thokra-nav) [reply pkg/resourcecreator/testdata/naisjob/vanilla_job_on_premises.yaml:126]: YAML doesn't really care about the number of indents I guess, just that it's consistent within each block

**Review by kimtore — APPROVED** [review]

**Pull request merged**